safe/impact_functions/ash/ash_raster_places/impact_function.py

```diff
@@ -175,6 +181,12 @@ def run(self):
 
             self.classify_feature(category, usage, population, True)
 
+        geometries = interpolated_layer.get_geometry()
+        unaffected_feats.reverse()
+        for u in unaffected_feats:
+            features.remove(features[u])
+            geometries.remove(geometries[u])
```

Comment:
track unaffected features and exclude it from impact layer. It doesn't make sense to include places in unaffected area.